packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -1,16 +1,108 @@
+
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+Amazon CloudWatch Synthetics allow you to monitor your application by generating **synthetic** traffic. The traffic is produced by a **canary**: a configurable script that runs on a schedule. You configure the canary script to follow the same routes and perform the same actions as a user, which allows you to continually verify your user experience even when you don't have any traffic on your applications.
+
+## Canary
+
+To illustrate how to use a canary, assume your application defines the following endpoint:
+
+```bash
+% curl "https://api.example.com/user/books/topbook/"
+The Hitchhikers Guide to the Galaxy
+
+```
+
+The below code defines a canary that will hit the `books/topbook` endpoint every 5 minutes: 
 
 ```ts
 import * as synthetics from '@aws-cdk/aws-synthetics';
+
+const canary = new synthetics.Canary(this, 'MyCanary', {
+  schedule: synthetics.Schedule.rate(Duration.minutes(5)),
+  test: Test.custom({
+    code: Code.fromAsset(path.join(__dirname, 'canary'))),
+    handler: 'index.handler',
+  }),
+});
+```
+
+The following is an example of a `canary/index.js` which exports the `handler` function:
+
+```
+const https = require('https');
+var synthetics = require('Synthetics');
+const log = require('SyntheticsLogger');
+
+exports.handler = async function () {
+  const requestOptions = {"hostname":"api.example.com","method":"","path":"/user/books/topbook/","port":443}
+  let req = https.request(requestOptions);
+  req.on('response', (res) => {
+    log.info()
+  });
+}
+```
+
+
+The canary will automatically produce a CloudWatch Dashboard:
+
+![UI Screenshot](images/ui-screenshot.png)
+
+### Configuring the Canary Script 
+
+To configure the script the canary executes, use the `test` property. The `test` property exposes `code` and `handler` properties -- both are required by Synthetics to create a lambda function on your behalf. 
+
+The `synthetics.Code` class exposes static methods to bundle your code artifacts: 
+
+  - `code.fromInline(code)` - specify an inline script.
+  - `code.fromAsset(path)` - specify a .zip file or a directory in the local filesystem which will be zipped and uploaded to S3 on deployment. See note for directory structure.
+  - `code.fromBucket(bucket, key[, objectVersion])` - specify an S3 object that contains the .zip file of your runtime code. See note for directory structure.
+
+> **Note:** For `code.fromAsset()` and `code.fromBucket()`, the canary resource requires the following folder structure: `nodejs/node_modules/<handlerFile>`. See Synthetics [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_WritingCanary.html).
```

Comment:
I would move this up just after the example of `index.js`. And use standard directory structure ASCII:

```text
canary/
├── nodejs/
    ├── node_modules/
        ├── index.js
```

Reply:
lovely